easy-to-use-api/klips-url-generator/src/components/PhotonSearch/index.tsx

```diff
@@ -0,0 +1,191 @@
+import React, {
+  useEffect,
+  useState
+} from 'react';
+
+import OlVectorLayer from 'ol/layer/Vector.js';
+import OlVectorSource from 'ol/source/Vector.js';
+import OlFormatGeoJSON from 'ol/format/GeoJSON';
+import * as olExtent from 'ol/extent';
+import { unByKey } from 'ol/Observable';
+
+import MapUtil from '@terrestris/ol-util/dist/MapUtil/MapUtil.js';
+import Logger from '@terrestris/base-util/dist/Logger';
+
+import useMap from '@terrestris/react-geo/dist/Hook/useMap';
+import { AutoComplete } from 'antd';
+import { style } from '../../constants'
+import './index.css';
+import { Feature, FeatureCollection, Geometry } from 'geojson';
+import { DefaultOptionType } from 'antd/lib/select';
+
+export type PhotonProperties = {
+  osm_id?: number;
+  extent?: [number, number, number, number];
+  country?: string;
+  county?: string;
+  city?: string;
+  postcode?: string;
+  type?: string;
+  osm_type?: string;
+  osm_key?: string;
+  district?: string;
+  osm_value?: string;
+  name?: string;
+  state?: string;
+  countrycode?: string;
+  locality?: string;
+  housenumber?: string;
+  street?: string;
+};
+
+export type PhotonFeatureCollection = FeatureCollection<Geometry, PhotonProperties>;
+
+export type PhotonOptionType = {
+  feature: Feature<Geometry, PhotonProperties>;
+} & DefaultOptionType;
+
+export const PhotonSearch: React.FC = (
+): JSX.Element => {
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [dataSource, setDataSource] = useState<PhotonFeatureCollection>();
+  const [bbox, setBbox] = useState<olExtent.Extent | undefined>();
+
+  const map = useMap();
+
+  useEffect(() => {
+    if (!map) {
+      return;
+    }
+
+    const bboxLayer = MapUtil.getLayerByName(map, 'BboxLayer') as OlVectorLayer<OlVectorSource>;
+    let bbox;
+
+    const key = bboxLayer.getSource()?.on('addfeature', olEvt => {
+      bbox = olEvt.feature?.clone().getGeometry()?.transform('EPSG:3857', 'EPSG:4326').getExtent();
+
+      if (!bbox) {
+        return;
+      }
+
+      setBbox(bbox)
+    });
+
+    return () => {
+      if (!key) {
+        return;
+      }
+
+      unByKey(key);
+    };
+  }, [map]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch(`https://photon.komoot.io/api?q=${searchTerm}&bbox=${bbox?.toString()}`);
```

Comment:
minor, put the url in a variable, so that it can be changed later easily (if we have an owned-hosted photon)